server/src/test/java/org/elasticsearch/index/shard/IndexShardTests.java

```diff
@@ -4552,6 +4554,144 @@ public void testRecordsForceMerges() throws IOException {
         closeShards(shard);
     }
 
+    public void testShardExposesWriteLoadStats() throws Exception {
```

Comment:
Can we supplement this with an integration test that in essence does something like:

```
assertThat(writeLoad == 0);
assertBusy(() -> {
  // index some docs
  assertThat(writeLoad > 0);
});
```

just to check that this works integrated too.